app/experimenter/projects/forms.py

```diff
@@ -33,7 +29,27 @@ def clean(self):
         return cleaned_data
 
 
-class ProjectForm(AutoNameSlugFormMixin, forms.ModelForm):
+class UniqueNameSlugFormMixin(NameSlugFormMixin):
+    """
+    Automatically generate a slug from the name field
+    and ensure that it is unique across all instances
+    """
+
+    def clean_name(self):
+        name = super().clean_name()
+        slug = slugify(name)
+
+        if (
+            self.instance.pk is None
```

Comment:
So does this mean that when an experiment's name is updated, it could be changed to something that would cause a collision?

ie:

1. Make an experiment named "A"
2. Make an experiment named "B"
3. Rename "A" to "B"

Since A will have a primary key now, this check would allow it, right?

Reply:
Yes, the regular Django model uniqueness validation catches that case, I had to add this additional check for the case where two names are not unique but their slugs collide before the experiment is saved.

Reply:
The other part of the answer here is that the slug is not automatically regenerated if the name changes (for now?).